src/jit/simdcodegenxarch.cpp

```diff
@@ -1440,6 +1440,57 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
     genConsumeOperands(simdNode);
     regNumber srcReg = op1->gtRegNum;
 
+    // Optimize the case of op1 is in memory and trying to access ith element.
+    if (op1->isMemoryOp())
+    {
+        assert(op1->isContained());
+
+        regNumber baseReg;
+        regNumber indexReg;
+        int       offset = 0;
+
+        if (op1->OperGet() == GT_LCL_FLD)
+        {
+            unsigned varNum = op1->gtLclVarCommon.gtLclNum;
+            offset += op1->gtLclFld.gtLclOffs;
+
+            bool isEBPbased;
+            offset += compiler->lvaFrameAddress(varNum, &isEBPbased);
```

Comment:
Very minor suggestion: To me it is more natural to set the offset to the local, and then add the offset of the field, though obviously it doesn't matter functionally, since addition is commutative. Also, a comment clarifying that there are 3 parts to the total offset might be useful, i.e. {offset of local} + {offset of SIMD Vector field} + {offset of element within SIMD Vector}.